nifi-nar-bundles/nifi-kafka-bundle/nifi-kafka-0-10-processors/src/main/java/org/apache/nifi/processors/kafka/pubsub/ConsumerLease.java

```diff
@@ -392,13 +394,27 @@ private void writeDemarcatedData(final ProcessSession session, final List<Consum
         bundleMap.put(topicPartition, tracker);
     }
 
+    private void rollback(final TopicPartition topicPartition) {
+        final OffsetAndMetadata offsetAndMetadata = kafkaConsumer.committed(topicPartition);
```

Comment:
instead of asking the kafka client where we were as far as committed offsets we should first check the uncommittedOffsets map and if nothing then ask the kafkaclient otherwise use the last uncommittedOffset for this topic/partition pair.